packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/handles/useHandleChange.tsx

```diff
@@ -54,7 +54,9 @@ export const useHandleChange = (): UseHandleChangeValue => {
                   sourcePosition,
                   targetPosition,
                   sourceHandle,
-                  targetHandle
+                  targetHandle,
+                  getNodes(),
+                  getEdges()
```

Comment:
For your information, the code below using `setNodes` is exactly similar to this:

```
setState(prevState => prevState.map(node => {
    node.data.connectionHandles = sourceConnectionHandles; // I'm simplifying
    return node;
});
```

You can't just modify the previous state like in a setState operation. @gcoutable is working on improving that.

Reply:
I don't see the issue, I'm doing it like in the [doc]( https://reactflow.dev/examples/nodes/update-node)
I don't know if it's relevant but please not that I also have a custom hook to refresh my handles `useRefreshConnectionHandles`

Reply:
To add to the discussion, if I look at the doc of [Zustand ](https://docs.pmnd.rs/zustand/guides/immutable-state-and-merging)used by ReactFlow. 
Quote : `However, as this is a common pattern, set actually merges state, and we can skip the ...state part:`